fem/intrules.hpp

```diff
@@ -28,6 +28,22 @@ class IntegrationPoint
    double x, y, z, weight;
    int index;
 
+   IntegrationPoint() = default;
+
+   IntegrationPoint (const double* p, int dim)
```

Comment:
I added custom constructors to create an `IntegrationPoint` from a `double*` and  a given dimension, `dim` and ` double*` and weight.  This was prior to the  introduction of the `index` member variable. 

I am assuming the `index` is according to the local numbering scheme of an element. Is this assumption correct?

How should we initialize this here? 
* Just set it to zero?
* Make it an optional argument to the new constructors, which defaults to zero if not specified?

For the particular use-case, I am not making use of the "index".

Reply:
Yes, `index` can be set in the context of an `IntegrationRule` according to a local numbering.

It is probably best to use `-1` for `index` to indicate invalid/undefined value, since `0` is a valid value.

Regarding the addition of an optional parameter in the constructor for the `index` -- either choice is fine -- use what you prefer.

Reply:
Sounds good --  thanks @v-dobrev